src/Microsoft.DotNet.XHarness.Android/AdbRunner.cs

```diff
@@ -653,67 +659,85 @@ public List<string> PullFiles(string apkPackageName, string devicePath, string l
 
     private IReadOnlyCollection<AndroidDevice> GetDevices(params AdbProperty[] propertiesToLoad)
     {
-        var devices = new List<AndroidDevice>();
+        string[] standardOutputLines = Array.Empty<string>();
+        ProcessExecutionResults result = null!;

```

Comment:
Yeah.. I wasn't sure how to make it so that it knows I am setting the value inside of the retry lambda.

I might need to make the `Retry` return a value and recognize `null` as a failed retry

Reply:
I've made it so that Retry returns the result of the action and it's a bit cleaner now